python/artm/model.py

```diff
@@ -0,0 +1,723 @@
+          batches (list of str): list of file names of batches to be processed;
+          if not None, than data_format should be 'batches'; format '*.batch',
+          default=None
+          data_path (str):
+          1) if data_format == 'batches' =>
+          folder containing batches and dictionary;
+          2) if data_format == 'bow_uci' =>
+          folder containing docword.txt and vocab.txt files;
+          3) if data_format == 'vowpal_wabbit' => file in Vowpal Wabbit format;
+          4) if data_format == 'plain_text' => file with text;
+          default=''
+          data_format (str): the type of input data;
+          1) 'batches' --- the data in format of BigARTM;
+          2) 'bow_uci' --- Bag-Of-Words in UCI format;
+          3) 'vowpal_wabbit' --- Vowpal Wabbit format;
+          4) 'plain_text' --- source text;
+          default='batches'
+
+        Returns:
+          pandas.DataFrame: (data, columns, rows), where:
+          1) columns --- the ids of documents, for which the Theta
+          matrix was requested
+          2) rows --- the names of topics in topic model, that was
+          used to create Theta
+          3) data --- content of Theta matrix.
+        """
+        if collection_name is None and data_format == 'bow_uci':
+            raise IOError('ArtmModel.find_theta(): No collection name was given')
+
+        if not data_format == 'batches' and batches is not None:
+            raise IOError("ArtmModel.find_theta(): batches != None require" +
+                          "data_format == batches")
+
+        if not self._initialized:
+            raise RuntimeError("ArtmModel.find_theta(): Model does not exist yet. Use " +
+                               "ArtmModel.initialize()/ArtmModel.fit_*()")
+
+        target_folder = data_path
+        if not data_format == 'batches':
+            target_folder = tempfile.mkdtemp()
+        try:
+            batches_list = _parse_collection_inline(target_folder,
+                                                    data_path,
+                                                    data_format,
+                                                    'find_theta',
+                                                    collection_name,
+                                                    batches,
+                                                    batch_size=1000)
+
+            results = self.master.ProcessBatches(
+                pwt=self.model,
+                batches=batches_list,
+                target_nwt='nwt_hat',
+                inner_iterations_count=self._num_document_passes,
+                class_ids=self._class_ids,
+                theta_matrix_type=lib.ProcessBatchesArgs_ThetaMatrixType_External)
+
+            document_ids = [item_id for item_id in results[0].theta_matrix.item_id]
+            topic_names = [topic_name for topic_name in results[0].theta_matrix.topic_name]
+            retval = DataFrame(data=results[1].transpose(),
```

Comment:
Awful convention!
`retval` says nothing about variable content.
It is obvious with `return` construction that the variable is a value that returns. But one cant even understand what the function value is. The code says: "this function returns the returning value", awesom.